Fix infer issues

philljj opened the pull request:
# Description

Fix `UNINITIALIZED_VALUE` bugs from infer testing.

# Testing

Confirmed fixed with Infer rerun.

philljj commented:
Retest this please

philljj commented:
Retest this please

douzzer commented on src/ssl.c:
```diff
@@ -30357,6 +30371,8 @@ static int set_curves_list(WOLFSSL* ssl, WOLFSSL_CTX *ctx, const char* names)
 #else
     int groups[WOLFSSL_MAX_GROUP_COUNT];
+
+    XMEMSET(groups, 0, sizeof(groups));
```
This is another expensive `memset` to address a clearly-false positive, as above. in `wolfSSL_CTX_set1_groups()` etc.

douzzer commented on src/ssl.c:
```diff
@@ -30366,6 +30382,8 @@ static int set_curves_list(WOLFSSL* ssl, WOLFSSL_CTX *ctx, const char* names)
         goto leave;
     }
+
+    XMEMSET(groups, 0, sizeof(int)*WOLFSSL_MAX_GROUP_COUNT);
```
Same story, expensive op to address a false positive.

douzzer commented on src/wolfio.c:
```diff
@@ -1409,6 +1409,9 @@ int wolfIO_DecodeUrl(const char* url, int urlSz, char* outName, char* outPath,
             i = 0;
             cur++;
+
+            XMEMSET(port, 0, sizeof(port));
```
This addresses a false positive, but I can live with it, given it's only 6 bytes.  Btw I probably would've resolved it by refactoring that neighborhood to convert the ASCII port to a `word32` without the unnecessary copying of the ASCII.

douzzer reviewed (changes requested):
Looks good -- just a few overheady analyzer coddlings to redo in a non-costly way.

Passes all the latest analyzers locally in the `wolfssl-multi-test.sh ... super-quick-check` suite, which is good news -- it's entirely possible to coddle one analyzer only to have a different analyzer report an unused value or other defect around the coddling.

philljj commented:
Rebased and removed the added `XMEMSET(groups` and `XMEMSET(_groups`. We are considering the `groups` to be initialized by the for loop logic and so these are false positives.

douzzer reviewed (approved):
LGTM
